### ISSUE TITLE
A question on Scripts/generate.py

### ISSUE THREAD

**Issue by ztcintokyo** [facebookresearch/access#12]
Hello!
I tried to use generate.py for sentence simplification. I use the command "python scripts/generate.py < my_file.complex", but neither could I find the generated files, nor any output.
In my_file.complex, there are just origin sentences one for each line.
Thank you for your reply!

**Comment by louismartin** [comment]
Hi, Thanks for raising the issue.
Could you please drop into pdb [here](https://github.com/facebookresearch/access/blob/1428b4ed3413a2a410f9aa4bbff54429c2110eff/scripts/generate.py#L20) by adding `import pdb; pdb.set_trace()`, and go step by step to find what's the problem?
Things you should investigate:
* Check that `write_lines([word_tokenize(line) for line in fileinput.input()], source_filepath)` has actually written the input lines to `source_filepath`
* check that `simplifier(source_filepath, pred_filepath)` wrote some lines in `pred_filepath`.
* You can also comment out the `with mute():` line to be sure no error message is suppresed.

Is that something you can do ?

**Comment by louismartin** [comment]
Or simply create a script `generate_debug.py` with the following content and give me the output of `python generate_debug.py  < my_file.complex`

```
# Copyright (c) Facebook, Inc. and its affiliates.
# All rights reserved.
#
# This source code is licensed under the license found in the
# LICENSE file in the root directory of this source tree.
#

import fileinput

from access.preprocessors import get_preprocessors
from access.resources.prepare import prepare_models
from access.simplifiers import get_fairseq_simplifier, get_preprocessed_simplifier
from access.text import word_tokenize
from access.utils.helpers import yield_lines, write_lines, get_temp_filepath, mute


if __name__ == '__main__':
    # Usage: python generate.py < my_file.complex
    # Read from stdin
    source_filepath = get_temp_filepath()
    lines = [word_tokenize(line) for line in fileinput.input()]
    print(f'Input lines: {lines[:3]}')
    write_lines(lines, source_filepath)
    with open(source_filepath, 'r') as f:
        lines = [line for line in f]
        print(f'Written lines: {lines[:3]}')
    # Load best model
    best_model_dir = prepare_models()
    recommended_preprocessors_kwargs = {
        'LengthRatioPreprocessor': {'target_ratio': 0.95},
        'LevenshteinPreprocessor': {'target_ratio': 0.75},
        'WordRankRatioPreprocessor': {'target_ratio': 0.75},
        'SentencePiecePreprocessor': {'vocab_size': 10000},
    }
    preprocessors = get_preprocessors(recommended_preprocessors_kwargs)
    simplifier = get_fairseq_simplifier(best_model_dir, beam=8)
    simplifier = get_preprocessed_simplifier(simplifier, preprocessors=preprocessors)
    # Simplify
    pred_filepath = get_temp_filepath()
    print(f'Pred filepath: {pred_filepath}')
    simplifier(source_filepath, pred_filepath)
    for line in yield_lines(pred_filepath):
        print(line)
```

**Comment by alexmomeni** [comment]
Hi @louismartin , I get this bug
"generate_debug.py: error: unrecognized arguments: --raw-text"
I actually got the same when trying to debug and run the script without mute() this weekend

**Comment by louismartin** [comment]
> Hi @louismartin , I get this bug
> "generate_debug.py: error: unrecognized arguments: --raw-text"
> I actually got the same when trying to debug and run the script without mute() this weekend

Hi @alexmomeni , your bug is most probably due to the wrong fairseq version being installed.
```
pip install --upgrade
pip install -r requirements.txt
```

**Comment by ztcintokyo** [comment]
Hi @louismartin,  I tried to use generate_debug.py but got the same error as @alexmomeni .
I upgrade the requirement again as you mentioned above, but didn't work.

**Comment by alexmomeni** [comment]
Hi @louismartin, tried upgrading but didn't work either on my end.

Here is the full error

usage: generate_debug.py [-h] [--no-progress-bar] [--log-interval N]
                         [--log-format {json,none,simple,tqdm}]
                         [--tensorboard-logdir DIR] [--seed N] [--cpu]
                         [--fp16] [--memory-efficient-fp16]
                         [--fp16-no-flatten-grads]
                         [--fp16-init-scale FP16_INIT_SCALE]
                         [--fp16-scale-window FP16_SCALE_WINDOW]
                         [--fp16-scale-tolerance FP16_SCALE_TOLERANCE]
                         [--min-loss-scale D]
                         [--threshold-loss-scale THRESHOLD_LOSS_SCALE]
                         [--user-dir USER_DIR]
                         [--empty-cache-freq EMPTY_CACHE_FREQ]
                         [--all-gather-list-size ALL_GATHER_LIST_SIZE]
                         [--model-parallel-size N]
                         [--checkpoint-suffix CHECKPOINT_SUFFIX]
                         [--criterion {cross_entropy,adaptive_loss,legacy_masked_lm_loss,nat_loss,label_smoothed_cross_entropy,composite_loss,binary_cross_entropy,sentence_prediction,label_smoothed_cross_entropy_with_alignment,masked_lm,sentence_ranking,vocab_parallel_cross_entropy}]
                         [--tokenizer {nltk,space,moses}]
                         [--bpe {sentencepiece,fastbpe,gpt2,subword_nmt,hf_byte_bpe,bert,byte_bpe,characters,bytes}]
                         [--optimizer {nag,adafactor,sgd,adamax,adagrad,adam,lamb,adadelta}]
                         [--lr-scheduler {fixed,reduce_lr_on_plateau,polynomial_decay,inverse_sqrt,tri_stage,cosine,triangular}]
                         [--task TASK] [--num-workers N]
                         [--skip-invalid-size-inputs-valid-test]
                         [--max-tokens N] [--max-sentences N]
                         [--required-batch-size-multiple N]
                         [--dataset-impl FORMAT] [--gen-subset SPLIT]
                         [--num-shards N] [--shard-id ID] [--path FILE]
                         [--remove-bpe [REMOVE_BPE]] [--quiet]
                         [--model-overrides DICT] [--results-path RESDIR]
                         [--beam N] [--nbest N] [--max-len-a N]
                         [--max-len-b N] [--min-len N] [--match-source-len]
                         [--no-early-stop] [--unnormalized] [--no-beamable-mm]
                         [--lenpen LENPEN] [--unkpen UNKPEN]
                         [--replace-unk [REPLACE_UNK]] [--sacrebleu]
                         [--score-reference] [--prefix-size PS]
                         [--no-repeat-ngram-size N] [--sampling]
                         [--sampling-topk PS] [--sampling-topp PS]
                         [--temperature N] [--diverse-beam-groups N]
                         [--diverse-beam-strength N] [--diversity-rate N]
                         [--print-alignment] [--print-step]
                         [--iter-decode-eos-penalty N]
                         [--iter-decode-max-iter N]
                         [--iter-decode-force-max-iter]
                         [--iter-decode-with-beam N]
                         [--iter-decode-with-external-reranker]
                         [--retain-iter-history]
                         [--decoding-format {unigram,ensemble,vote,dp,bs}]
                         [--momentum M] [--weight-decay WD] [--force-anneal N]
                         [--lr-shrink LS] [--warmup-updates N] [-s SRC]
                         [-t TARGET] [--load-alignments]
                         [--left-pad-source BOOL] [--left-pad-target BOOL]
                         [--max-source-positions N] [--max-target-positions N]
                         [--upsample-primary UPSAMPLE_PRIMARY]
                         [--truncate-source] [--eval-bleu]
                         [--eval-bleu-detok EVAL_BLEU_DETOK]
                         [--eval-bleu-detok-args JSON] [--eval-tokenized-bleu]
                         [--eval-bleu-remove-bpe [EVAL_BLEU_REMOVE_BPE]]
                         [--eval-bleu-args JSON] [--eval-bleu-print-samples]
                         data
generate_debug.py: error: unrecognized arguments: --raw-text

**Comment by louismartin** [comment]
Can you print `pip freeze | grep fairseq` ?

**Comment by ztcintokyo** [comment]
-e git+https://github.com/pytorch/fairseq.git@fb76dac1c4e314db75f9d7a03cb4871c532000cb#egg=fairseq

**Comment by alexmomeni** [comment]
> Can you print `pip freeze | grep fairseq` ?
fairseq==0.9.0

**Comment by louismartin** [comment]
Thanks for the reactivity. It is indeed a fairseq version problem, but I don't understand why `pip install -r requirements.txt` does not install the correct version.

In the meantime I think `pip install force-reinstall fairseq@git+https://github.com/louismartin/fairseq.git@controllable-sentence-simplification` will do the trick. Can you please try this command?

The output of `pip freeze | grep fairseq` should be `fairseq==0.6.2`.

**Comment by alexmomeni** [comment]
Thank you so much! This now works :) !

**Comment by louismartin** [comment]
Just out of curiosity, what's your use case for using ACCESS? :)

**Comment by ztcintokyo** [comment]
Thank you for your quick reply!

Just out of curiosity, what's your use case for using ACCESS? :)
Well, just curious about the SOTA of simplification, nice work!

**Comment by louismartin** [comment]
I'll close the issue, feel free to re-open if you have other problems :)
